File: src/main/java/ua/softgroup/matrix/server/persistent/repository/WorkTimePeriodRepository.java
package ua.softgroup.matrix.server.persistent.repository;

import org.springframework.data.repository.CrudRepository;
import ua.softgroup.matrix.server.persistent.entity.WorkDay;
import ua.softgroup.matrix.server.persistent.entity.WorkTimePeriod;

public interface WorkTimePeriodRepository extends CrudRepository<WorkTimePeriod, Long> {

    WorkTimePeriod findTopByWorkDayOrderByStartAsc(WorkDay workDay);

}

File: src/main/scala/ua/softgroup/matrix/server/persistent/repository/WorkTimePeriodRepository.scala
package ua.softgroup.matrix.server.persistent.repository

import org.springframework.data.repository.CrudRepository
import ua.softgroup.matrix.server.persistent.entity.{WorkDay, WorkTimePeriod}

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com> 
  */
trait WorkTimePeriodRepository extends CrudRepository[WorkTimePeriod, java.lang.Long] {

  def findTopByWorkDayOrderByStartAsc(workDay: WorkDay): WorkTimePeriod

  def findTopByWorkDayOrderByEndDesc(workDay: WorkDay): WorkTimePeriod

  def findTopByWorkDayOrderByStartDesc(workDay: WorkDay): WorkTimePeriod

}
